.github/workflows/scriptcheck.yml

```diff
@@ -154,9 +154,21 @@ jobs:
         env:
           PYTHONPATH: ./addons
 
-      - name: test htmlreport
+      - name: test htmlreport (standalone)
         run: |
-          htmlreport/test_htmlreport.py
+          htmlreport/tests/test_htmlreport.py
```

Comment:
While we are moving stuff. Is it possible (I mean in terms of how the test is written/executed and without some reworking) to move this under `test/tools` (or similar)?

We currently have different encapsulation strategies. In some cases everything belonging to component is included in the same folder and in some cases it is split logically. (Or no strategy at all in case of the `tools` folder which is basically still a collection of random things).

In terms of easing the packaging (IIRC we unintentionally packaged internal stuff in the past) and testing I would prefer if things are logically split i.e. all tests go under `test` and the respective folder only contains the production code.

Reply:
Would like to confirm with you:
- do you mean `test` folder in the root of the repository?

If so, yes it should be possible to update the directory without affecting the existing test implementation.

Reply:
It seems like we also need to move all supplementary files like `example.cc`, `example.xml`, etc. Would you tell if you expect them to be moved as well or you prefer some different approach?

Reply:
Extracted to the `test/tools` and updated directories inside the test suite.

Reply:
Please add a `htmlreport` subfolder otherwise it would just add up to the pile of "random" looking things.

Also we should move all test-related files including `check.sh`, `test_htmlreport.py` and `test_suppressions.txt`.

We should not rename the test file though. Otherwise it might be automatically picked by `pytest` and we (at least currently) do not want that for this test.

Reply:
Ah. The actual test was already moved. I missed that because there were also content changes in addition to the move.

Reply:
So, do you want me to create a subdir called `htmlreport` inside the `test/tools` and move previously moved files there?

Reply:
Yes, please.

Reply:
Updated according to the request.